source/Octopus.Client.Tests/Exceptions/OctopusExceptionFactoryFixture.cs

```diff
@@ -0,0 +1,304 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using Octopus.Client.Exceptions;
+
+namespace Octopus.Client.Tests.Exceptions
+{
+    public class OctopusExceptionFactoryFixture
+    {
+        [Test]
+        [TestCase(HttpStatusCode.BadRequest)]
+        [TestCase(HttpStatusCode.Conflict)]
+        public async Task Http400or409Response_WithNoPayLoad_ShouldCreateOctopusValidationException(
+            HttpStatusCode statusCode)
+        {
+            var httpResponseMessage = new HttpResponseMessage(statusCode);
+            var createdException = await OctopusExceptionFactory.CreateException(httpResponseMessage);
+
+            using (new AssertionScope())
+            {
+                createdException.Should().BeOfType<OctopusValidationException>();
+                createdException.HttpStatusCode.Should().Be((int)statusCode);
+            }
+        }
+
+        [Test]
+        [TestCaseSource(nameof(CreatedOctopusExceptionTestData))]
+        public async Task CreatedOctopusValidationException_ShouldContainPayloadInformation(
+            OctopusExceptionFactory.OctopusErrorsContract octopusErrorsContract)
+        {
+            var httpResponseMessage = new HttpResponseMessage(HttpStatusCode.BadRequest);
+
+            OctopusExceptionFactory.OctopusErrorsContract resultValidationObject = null;
+            if (HasAnyData(octopusErrorsContract))
+            {
+                var content = JsonConvert.SerializeObject(octopusErrorsContract);
+                httpResponseMessage.Content = new StringContent(content);
+                resultValidationObject = OctopusExceptionFactory.OctopusErrorsContractFromBody(content);
+            }
+
+            var createdException =
+                await OctopusExceptionFactory.CreateException(httpResponseMessage) as OctopusValidationException;
+
+            using (new AssertionScope())
+            {
+                createdException.Should().NotBeNull();
```

Comment:
Minor nitpick, but this assertion needn't be inside the scope, given if it fails there's no point continuing with the others.